fastcrypto-zkp/src/bn254/utils.rs

```diff
@@ -15,27 +14,45 @@ use serde::Deserialize;
 use serde_json::json;
 use std::str::FromStr;
 
-use super::zk_login::ZkLoginInputs;
+use super::zk_login::{hash_ascii_str_to_field, to_field};
 
 const ZK_LOGIN_AUTHENTICATOR_FLAG: u8 = 0x05;
 const SALT_SERVER_URL: &str = "http://salt.api-devnet.mystenlabs.com/get_salt";
-const PROVER_SERVER_URL: &str = "http://185.209.177.123:8000/test/zkp";
+const PROVER_SERVER_URL: &str = "http://185.209.177.123:7000/zkp";
+const MAX_KEY_CLAIM_NAME_LENGTH: u16 = 32;
+const MAX_KEY_CLAIM_VALUE_LENGTH: u16 = 115;
+const MAX_AUD_VALUE_LENGTH: u16 = 145;
 
 /// Calculate the Sui address based on address seed and address params.
-pub fn get_zk_login_address(
-    address_seed: &str,
-    param: AddressParams,
-) -> Result<[u8; 32], FastCryptoError> {
+pub fn get_zk_login_address(address_seed: &str, iss: &str) -> Result<[u8; 32], FastCryptoError> {
     let mut hasher = Blake2b256::default();
     hasher.update([ZK_LOGIN_AUTHENTICATOR_FLAG]);
-    hasher.update(
-        bcs::to_bytes(&AddressParams::new(param.iss, param.aud))
-            .map_err(|_| FastCryptoError::InvalidInput)?,
-    );
+    let bytes = iss.as_bytes();
+    hasher.update([bytes.len() as u8]);
```

Comment:
let's verify that len() fits in u8

Reply:
yes - MAX_EXT_ISS_LEN = 165 fixed above < u8::max = 255